### PR TITLE
doc: src/Lean/Declaration.lean

### PR DESCRIPTION
**Pull request by bollu**
Add details explained by Leo at the hackathon.

### CODE REVIEW

File: src/Lean/Declaration.lean [hunk 1]
```diff
@@ -213,6 +220,19 @@ structure InductiveVal extends ConstantVal where
   isRec : Bool
   /-- Whether the definition is flagged as unsafe. -/
   isUnsafe : Bool
+  /-- An of inductive type called reflexive if it has at least one constructor that takes as an argument a function returning the
```
Comment (semorrison) [inline src/Lean/Declaration.lean:223]:
```suggestion
  /-- An inductive type is called reflexive if it has at least one constructor that takes as an argument a function returning the
```

File: src/Lean/MetavarContext.lean [hunk 2]
```diff
@@ -252,7 +256,13 @@ structure MetavarDecl where
   userName       : Name := Name.anonymous
   lctx           : LocalContext
   type           : Expr
-  depth          : Nat
+  depth          : Nat /--
+    The nesting depth of this metavariable. We do not want a
```
Comment (semorrison) [inline src/Lean/MetavarContext.lean:260]:
```suggestion
    The nesting depth of this metavariable. We do not want
```

File: src/Lean/MetavarContext.lean [hunk 3]
```diff
@@ -252,7 +256,13 @@ structure MetavarDecl where
   userName       : Name := Name.anonymous
   lctx           : LocalContext
   type           : Expr
-  depth          : Nat
+  depth          : Nat /--
```
Comment (leodemoura) [inline src/Lean/MetavarContext.lean:259]: This docstring must be before `depth : Nat`

**Comment by leodemoura** [comment]
@bollu Thanks for submitting the doc-string PRs. They require minor changes, are you going to have time to do it?

Thanks,
Leo

**Comment by bollu** [comment]
Yes, I will. I'm currently travelling, and I'll be ready to fixup my
patches on Wednesday.


Best,
~Siddharth

File: src/Lean/MetavarContext.lean (hunk 3 above)
Comment (bollu) [reply src/Lean/MetavarContext.lean]: Done!

**Pull request merged**